pushnotifications/build.gradle

```diff
@@ -32,8 +32,10 @@ dependencies {
     implementation 'com.squareup.retrofit2:converter-gson:2.3.0'
     implementation 'com.squareup.retrofit2:retrofit:2.5.0'
     implementation 'com.squareup.tape2:tape:2.0.0-beta1'
-    implementation 'com.squareup.moshi:moshi:1.8.0'
-    implementation 'com.squareup.moshi:moshi-adapters:1.8.0'
+    implementation 'com.squareup.moshi:moshi-kotlin:1.9.2'
+    implementation 'com.squareup.moshi:moshi-adapters:1.9.2'
+    implementation "dev.zacsweers.moshisealed:moshi-sealed-reflect:0.1.0"
+    implementation "dev.zacsweers.moshisealed:moshi-sealed-annotations:0.1.0"
```

Comment:
Not sure if the solution shouldn't be not to marshal a sealed class if it's not supported by Moshi... That extension lib created by one of the contributors is [WIP even when it comes to its API](https://www.reddit.com/r/androiddev/comments/dpjoy4/moshisealed_reflective_and_code_gen/f5vwsgn/). Also it seems like even with that lib, we need to remember about a lot of boiler-plate to put at each subclass of the sealed class.

Alternative solution could be about separately persisting given types of `ServerSyncJob`s so that we always know which type to expect when unmarshalling.

Reply:
I think this is a question that @luismfonseca could help answer - are we okay using a WIP library to mitigate using sealed classes with moshi, or we could spend some time refactoring this piece of code to explicitly save the classes and not use a sealed type?

Reply:
From my side that comment is more to serve as lesson learned for new code, sorry for not expressing it clearly. I don't think it would be a good ROI in doing a broader and tricky refactoring here.

Reply:
> are we okay using a WIP library to mitigate using sealed classes with moshi, or we could spend some time refactoring this piece of code to explicitly save the classes and not use a sealed type?

I think if we do sufficient testing on our end to make sure the library works well for us, and if we are happy with how it's performing, then we can use it. The fact that's not an official library means that we need to put in more effort to make sure it works for us.

Also, I think that ideally we have the discussions about the way to solve something should happen before a PR -- not always possible, but how we go about solving something should be discussed with the potential reviewers, minimizing the late "this is better if implemented in this completely different way".

Reply:
Yeah, that sounds fair! I'll test on the sample app again to be 100% certain this is safe for us to use.

Totally agree with discussing approaches before PRs - in future I'll set up a huddle before finalising my approach to discuss any hesitations anyone may have before starting the work. Does that work?